### ISSUE TITLE
False positive type mismatch on unsizing structs

### ISSUE THREAD

**Issue by HKalbasi** [rust-lang/rust-analyzer#15133]
```Rust
#[derive(Debug)]
struct MyWrapper<T: ?Sized>(T);

fn x() {
    let x = &MyWrapper([1u8, 2, 3]);
    let y: &MyWrapper<[u8]> = x; // expected MyWrapper<[u8]> found MyWrapper<[u8; 3]>
}
```

**Comment by Veykril** [comment]
I think this is https://github.com/rust-lang/rust-analyzer/issues/13864?

**Comment by HKalbasi** [comment]
Yes that's it.
